guides/shipping.md

```diff
@@ -0,0 +1,126 @@
+# Shipping
+
+_Ahoy, matey!_
+Oh! you've built the greatest app ever and, of course, using Crystal!!
+And now you want to share it with the whole world?!
+
+Well, it’s your lucky ~~pirate~~ day because we are going to ship our first Crystal application!!
+
+So, _weigh anchor and hoist the mizzen! Yarr!_
+
+## The App
+
+The application we are shipping is an example of a static file sharing server. Here’s the source code:
+
+```crystal
+# staticserver.cr
+require "http"
+require "option_parser"
+
+# Handle Ctrl+C and kill signal.
+# Needed for hosting this process in a docker
+# as the entry point command
+Signal::INT.trap { puts "Caught Ctrl+C..."; exit }
+Signal::TERM.trap { puts "Caught kill..."; exit }
+
+path = "/www"
+port = 80
+
+option_parser = OptionParser.parse do |parser|
+ parser.on "-f PATH", "--files=PATH", "Files path (default: /www)" do |files_path|
+   path = files_path
+ end
+ parser.on "-p PORT", "--port=PORT", "Port to listen (default: 80)" do |server_port|
+   port = server_port.to_i
+ end
+end
+
+server = HTTP::Server.new([
+ HTTP::LogHandler.new,
+ HTTP::ErrorHandler.new,
+ HTTP::StaticFileHandler.new(path),
+])
+
+address = server.bind_tcp "0.0.0.0", port
+puts "Listening on http://#{address} and serving files in path #{path}"
+server.listen
+```
+
+So, starting the server (listening on port 8080 and serving files under the current directory) is as easy as running:
+
+```shell-session
+$ crystal ./src/staticserver.cr -- -p 8080 -f .
+Listening on http://0.0.0.0:8080 and serving files in path .
+```
+
+**Note:** the default behavior is to listen on `port 80` and serve the folder `/www`.
+
+## Compiling our application
+
+Let’s go over Crystal’s [introduction](https://crystal-lang.org/reference/). One of the  main goals of the language is to _Compile to efficient native code_. That means that each time that we compile our code then an executable is built, but with an important property: it has a target platform (architecture and operating system), which is where the application will run. Crystal knows the target platform because is the same as the one being used to compile.
+For example, if we use a Linux OS based computer for compiling, then the executable will be meant to run on a Linux OS (and in some cases we will need to use the same Linux distribution).
+
+Can we set the target when calling the compiler? Oh, that’s a great idea, but for now it’s not an option (there are a lot of great buccanears [working on a solution](https://forum.crystal-lang.org/t/cross-compiling-automatically-to-osx/1330/12) and remember that Crystal is open source: so you are welcome aboard!)
+
+Let’s compile our application:
+
+```shell-session
+$ shards build --production
```

Comment:
This is not about `shards`, so it rather only use `crystal build`.

Reply:
I like that shards create a nicer directory structure.

But either we show the shards.yml and the structure of the whole folder.
Or we use `crystal build` (and update the samples below removing the `bin`).

But since we are simulating a whole app, using shards makes sense to me. Otherwise, it feels we are not using our own tools.